src/dynarec/la64/dynarec_la64_660f.c

```diff
@@ -845,6 +954,28 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
                     u8 = F8;
                     VINSGR2VR_B(q0, ed, (u8 & 0xf));
                     break;
+                case 0x21:
+                    INST_NAME("INSERTPS Gx, Ex, Ib");
+                    nextop = F8;
+                    GETGX(q0, 1);
+                    if (MODREG) {
+                        q1 = sse_get_reg(dyn, ninst, x1, (nextop & 7) + (rex.b << 3), 0);
+                        u8 = F8;
+                        VEXTRINS_W(q0, q1, ((u8>>6)&3)|((u8>>4)&3) );
+                    } else {
+                        SMREAD();
+                        addr = geted(dyn, addr, ninst, nextop, &wback, x3, x5, &fixedaddress, rex, NULL, 0, 1);
+                        u8 = F8;
+                        LDxw(x2, wback, fixedaddress);
+                        VINSGR2VR_W(q0, x2, (u8>>4)&3);
+                    }
+                    uint8_t zmask = u8 & 0xf;
+                    for (uint8_t i=0; i<4; i++) {
+                        if (zmask & (1<<i)) {
+                            VINSGR2VR_W(q0, wZR, i);
+                        }
+                    }
```

Comment:
Please avoid using `VINSGR2VR_W` or similar as much as possible, such instructions have bad IPC.

Reply:
got it.